app/decorators/models/solidus_product_assembly/spree/product_decorator.rb

```diff
@@ -27,6 +27,15 @@ def self.prepended(base)
         end
       end
 
+      # Removes assembly parts whose variants have been deleted and includes
+      # part and that products part to be used in the parts_table.
+      def display_assembly_parts
+        assemblies_parts
+          .joins(:part)
+          .where(spree_variants: { deleted_at: nil })
```

Comment:
I think you can use `.merge(Spree::Variant.kept)` to hide this implementation detail (how records are soft-deleted).

Reply:
Somewhere along the way this is being added by rails for us. This line can be deleted.

```sql
> puts product.assemblies_parts.joins(:part).includes(part: :product).to_sql

SELECT "spree_assemblies_parts".*
FROM "spree_assemblies_parts"
INNER JOIN "spree_variants"
  ON "spree_variants"."id" = "spree_assemblies_parts"."part_id"
  AND "spree_variants"."deleted_at" IS NULL
WHERE "spree_assemblies_parts"."assembly_id" = 8698
ORDER BY "spree_assemblies_parts"."position" ASC
```